### ISSUE TITLE
Login Popup Redirecting Within Popup Rather Than Closing

### ISSUE THREAD

**Issue by SamP-Help4U** [AzureAD/microsoft-authentication-library-for-js#6675]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

3.1.0

### Wrapper Library

MSAL React (@azure/msal-react)

### Wrapper Library Version

2.0.3

### Public or Confidential Client?

Public

### Description

When performing a login via popup, after completing the login the popup won't close and will instead redirect to the redirect URL.
As a result, the application never receives the Authentication Result.
The issue is very intermittent and may not occur at all or may happen 10+ login attempts in a row.

![image](https://github.com/AzureAD/microsoft-authentication-library-for-js/assets/148498701/cc594f37-2e73-476f-9790-c20f47103690)

### Error Message

The issue does not throw any errors in the browser or application.

### MSAL Configuration

```javascript
{
    auth: {
        clientId: "clientId",
        authority: "https://login.microsoftonline.com/common",
        redirectUri: "http://localhost:3000",
        postLogoutRedirectUri: "/"
    },
    system: {
        allowNativeBroker: false // Disables WAM Broker
    }
}
```

### Relevant Code Snippets

```javascript
I was able to reproduce this issue on the MSAL react Typescript sample found here: 
https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/typescript-sample
```

### Reproduction Steps

1. Follow the 'How to run the sample' instructions in the typescript sample readme
2. Once the sample is running, click the "Login" button and select "Sign in using Popup"
3. Follow the auth flow
4. Once the auth flow is completed the popup will remain and within the popup will redirect to the specified redirect url
5. Steps 2-4 may need to be repeated several times before the issue occurs

### Expected Behavior

The popup should close after the auth flow is complete and pass the Authentication Result to the application.

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Chrome, Edge

### Source

External (Customer)

**Comment by tnorling** [comment]
This has been fixed in 3.5.0 - please upgrade

**Comment by cassioutfpr** [comment]
Hi, I'm facing the very same issue in my vue2 app.

Version: "@azure/msal-browser": "2.38.4"

Which already contains the fix for v3.5.0

**Comment by cassioutfpr** [comment]
I was able to fix this in my Vue 2 app. My router was set to redirect unauthenticated users to  `/login` , but Azure was configured to redirect them to  `/` (home), which caused the loop. I updated the Azure configuration to use redirectUri: window.location.origin + '/login' and added this route in Azure. Now everything works perfectly!
